### PR TITLE
Add working directory configuration option for Maven plugin run/start

### PR DESCRIPTION
**Pull request by plamentotev**
Add configuration option that specifies the working directory of the application ran by Maven plugin's run/start goals.

See #6243

### CODE REVIEW

**Comment by pivotal-issuemaster** [comment]
@plamentotev Please sign the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-projects/spring-boot&pullRequestId=6645)!

[Click here](https://cla.pivotal.io/sync/spring?repositoryId=spring-projects/spring-boot&pullRequestId=6645) to manually synchronize the status of this Pull Request.

See the [FAQ](https://cla.pivotal.io/faq) for frequently asked questions.

File: spring-boot-tools/spring-boot-loader-tools/src/main/java/org/springframework/boot/loader/tools/RunProcess.java [hunk 1]
```diff
@@ -41,12 +42,19 @@
...
 
 	private volatile long endTime;
 
+	public RunProcess(File workingDirectory, String... command) {
+		this.workingDirectory = workingDirectory;
+		this.command = command;
+	}
+
 	public RunProcess(String... command) {
 		this.command = command;
```
Comment (snicoll) [inline spring-boot-tools/spring-boot-loader-tools/src/main/java/org/springframework/boot/loader/tools/RunProcess.java]: This should call the other constructor

File: spring-boot-tools/spring-boot-loader-tools/src/main/java/org/springframework/boot/loader/tools/RunProcess.java [hunk 2]
```diff
@@ -58,6 +66,7 @@ public int run(boolean waitForProcess, String... args) throws IOException {
 	protected int run(boolean waitForProcess, Collection<String> args)
 			throws IOException {
 		ProcessBuilder builder = new ProcessBuilder(this.command);
+		builder.directory(workingDirectory);
```
Comment (snicoll) [inline spring-boot-tools/spring-boot-loader-tools/src/main/java/org/springframework/boot/loader/tools/RunProcess.java]: I wouldn't set it when it's `null`

**Comment by snicoll** [comment]
@plamentotev thanks for the PR, I've made a couple of comments.

File: spring-boot-tools/spring-boot-loader-tools/src/main/java/org/springframework/boot/loader/tools/RunProcess.java (hunk 2 above)
Comment (plamentotev) [inline spring-boot-tools/spring-boot-loader-tools/src/main/java/org/springframework/boot/loader/tools/RunProcess.java]: Actually `null` is valid value. It does nothing but still I feel a bit uncomfortable to just ignore the `null` value. In general I think in similar cases is better to no accept null values at all(if we consifer then as not a valid values) and throw NPE at the constructor. But I guess in this particular case it's enough just to add a note to the javadoc that the `null` for `wokingDirecotry` does not change the working directory.
What do you think?

Comment (snicoll) [inline spring-boot-tools/spring-boot-loader-tools/src/main/java/org/springframework/boot/loader/tools/RunProcess.java]: Now that I've looked at the javadoc of `ProcessBuillder#directory` I take it back: it is very explicit about the semantic of `null` there so I'd keep it like that. I'd add a javadoc on the constructor to reflect that information (but I can take care of that myself when I merge).

Feel free to update this PR, just push an additional commit on your branch (or preferably, rebase + push force)

**Comment by plamentotev** [comment]
@snicoll Thanks. As this is my first PR to the Spring project I have a question. The issue is targeted for 1.5.0 but I can't see a branch for this version and the master is 1.4.1 so how features for 1.5.0 are developed?

**Comment by snicoll** [comment]
They aren't. Master is `1.4` and we'll create a `1.4.x` at some point and master will become the branch for 1.5 work.

**Comment by pivotal-issuemaster** [comment]
@plamentotev Thank you for signing the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-projects/spring-boot&pullRequestId=6645)!

File: spring-boot-tools/spring-boot-loader-tools/src/main/java/org/springframework/boot/loader/tools/RunProcess.java (hunk 2 above)
Comment (plamentotev) [inline spring-boot-tools/spring-boot-loader-tools/src/main/java/org/springframework/boot/loader/tools/RunProcess.java]: I've updated the PR (+rebase). Also I've tried to add some javadoc comments. Writing comments (especially in English) is not my forte, so fell free to edit them :smile:

**Pull request closed without merging**